Makefile

```diff
@@ -28,28 +28,33 @@ GITHOOKS=.git/hooks
 DEPENDENCIES=$(EIGEN) $(RAPIDJSON) $(GITHOOKS)
 
 ifeq ($(CROSS), 0)
-	ifneq ($(UNAME),Linux)
-		# assume Windows
-		UNAME='Windows'
-		CFLAGS:=$(CFLAGS) -DWIN -std=gnu++11
-	else
-		CFLAGS:=$(CFLAGS) -std=c++11
-	endif
 	ARCH=$(shell uname -m)
 	ifneq (,$(findstring 64,$(ARCH)))
 		CFLAGS:=$(CFLAGS) -DARCH_64
 	endif
+else ifeq ($(UNAME), Darwin)
+    $(info cross-compiling using Mac master race host)
+	CC:=mac-crosscompiler/prebuilt/bin/clang++
+	export COMPILER_PATH = mac-crosscompiler/sysroot/usr/lib/gcc/arm-linux-gnueabihf/6.3.0/
+	CFLAGS:=$(CFLAGS) --target=arm-linux-gnueabihf \
+					  --sysroot=mac-crosscompiler/sysroot \
```

Comment:
Is this additional stuff necessary? Is there no g++ cross compiler? Could you perhaps symlink this to `hyped-cross-g++` to make it consistent with other cross-compilers?

Reply:
I tried looking into a g++ cross compiler, but setting it up seems very complex (that isn't the precompiled linux one) and using clang is relatively simple as it's natively a crosscompiler (some flags just have to be specified as you can see). I was thinking of updating the linux crosscompiler to use clang as well in the future.

Reply:
What does "it's natively a crosscompiler" mean?

Reply:
Instead of having to compile gcc for a specific host/target combination, clang [supports compiling to any target](https://clang.llvm.org/docs/CrossCompilation.html) out of the box.

Reply:
Previously, setting up cross compiler on Windows or Linux never involved compiling the gcc from scratch. We just installed one online, similarly how this PR installs the clang.

Reply:
Yes, but that's only because we were lucky someone had already compiled gcc targeting the raspberry pi for x86 hosts, we simply downloaded those files and used that. This uses a precompiled version of clang (available on their website to download) for mac hosts, but had to compile binutils from source unfortunately. I also copied over the beaglebone include and library files.

Reply:
We used the same for MAC, how about this one? https://s3.amazonaws.com/jaredwolff/xtools-2016-09-01.dmg


I actually do not mind either way, as I will probably never use this. It just seems like it would be much better if every platform used almost the same setup, and then the same Makefile script relevant to cross-compilation. This way, just be prepared I will give no support for cross-compilation issues to MAC users. I will also take every opportunity to make fun of MAC users is things do not work, as usual :)

Reply:
That's the file I meant by someone precompiled gcc. I agree it'd be nice if the same crosscompiler tools could be used for mac and linux, but at the end of the day there are inevitable differences between the two and if not now, then later there might be slight changes required to one platform to support it while keeping things the same for the other anyways.

I'll take responsibility for maintaining and supporting the mac crosscompiler :)